File: week-1/mario.c
#include <stdio.h>
#include <cs50.h>

void pyramid(int h, int total);

int main(void)
{
    int h = get_int("Height: ");
    while (h < 1 || h > 8)
    {
        h = get_int("Height: ");
    }
    pyramid(h, h);
}

void pyramid(int h, int total)
{
    if (h == 0)
        return;

    pyramid(h - 1, total);

    for (int i = 1; i <= total - h; i++)
    {
        printf(" ");
    }

    for (int i = 1; i <= h; i++)
    {
        printf("#");
    }

    printf("  ");

    for (int i = 1; i <= h; i++)
    {
        printf("#");
    }
    printf("\n");
}

File: week-6/mario.py
from cs50 import get_int


def main():
    height = get_int("Height: ")
    while height < 1 or height > 8:
        height = get_int("Height: ")
    print_pyramid(height)


def print_pyramid(height, original_height=None):
    if original_height == None:
        original_height = height
    if height == 0:
        return

    print_pyramid(height-1, original_height)
    blocks = ""

    # print left pad
    print("".rjust(original_height - height), end="")
    # concatenate blocks
    for i in range(height):
        blocks += "#"
    print(f"{blocks}  {blocks}")


if __name__ == "__main__":
    main()
